lib/clarkson-core-templates.php

```diff
@@ -395,65 +302,11 @@ public function get_templates( $choices = array() ) {
 		if ( $templates ) {
 			return $templates;
 		}
-		// Retrieve the cache list.
-		// If it doesn't exist, or it's empty prepare an array.
-		$theme = wp_get_theme();
-
-		if ( method_exists( $theme, 'get_page_templates' ) ) {
-			if ( version_compare( get_bloginfo( 'version' ), '4.7', 'lt' ) ) { // 4.6 and older
-				$templates = $theme->get_page_templates();
-			} else { // 4.7+
-				$templates = array();
-			}
-			if ( empty( $templates ) ) {
-				$templates = array();
-			}
-		} else {
-			$templates = array();
-		}
-
-		$templates      = array_merge( $templates, $choices );
+		$templates      = $choices;
 		$page_templates = array();
-		foreach ( $this->templates as $name => $path ) {
-			$is_valid_template = false;
-
-			/**
-			 * IF
-			 * Check if template matches of page-xyz.twig and skip page.twig
-			 * ELSE IF
-			 * Check for template-xyz.twig files and skip template.twig
-			 *
-			 * @since 0.2.1.
-			 */
-			if ( preg_match( '#^page-#i', $name ) === 1 && 'page' !== $name ) {
-				$is_valid_template = true;
-				$name              = str_replace( 'page-', '', $name );
-
-				/**
-				 * Allows turning of the warning for page- style naming, deprecated in 0.2.1.
-				 *
-				 * @hook clarkson_core_deprecated_warning_page_template
-				 * @since 0.2.1
-				 * @param {bool} WP_DEBUG WordPress debug mode constant.
-				 * @return {bool} Whether to show or hide the deprecation warning.
-				 *
-				 * @example
-				 * // Disable old style page warning.
-				 * add_filter( 'clarkson_core_deprecated_warning_page_template', '__return_false' );
-				 */
-				/**
-				 * @psalm-suppress UndefinedConstant
-				 */
-				$show_warning = apply_filters( 'clarkson_core_deprecated_warning_page_template', WP_DEBUG );
-				if ( $show_warning ) {
-					_doing_it_wrong( __METHOD__, 'Deprecated template name ' . esc_html( $path ) . ' found. Use `template-' . esc_html( $name ) . '.twig` instead.', '0.2.1' );
-				}
-			} elseif ( preg_match( '#^template-#i', $name ) === 1 && 'template' !== $name ) {
-				$is_valid_template = true;
-				$name              = str_replace( 'template-', '', $name );
-			}
-
-			if ( $is_valid_template ) {
+		foreach ( $this->get_template_files() as $name => $path ) {
+			if ( preg_match( '#^template-#i', $name ) === 1 && 'template' !== $name ) {
```

Comment:
`&& 'template' !== $name` is not required anymore. The preg_match will not return 1 otherwise.